RedacteurPortaal.Tests/Grains/ArchiveGrainTests.cs

```diff
@@ -41,10 +42,56 @@ public async Task CanAddArchiveCorrectly()
         Assert.Equal(toSaveArchive.MediaVideoItems, updatedArchive.MediaVideoItems);
         Assert.Equal(toSaveArchive.MediaPhotoItems, updatedArchive.MediaPhotoItems);
         Assert.Equal(toSaveArchive.Scripts, updatedArchive.Scripts);
+    }
 
+    [Fact]
+    public async Task CanGetASingularArchiveCorrectly()
+    {
+        var guid = Guid.NewGuid();
 
+        var toSaveArchive = new ArchiveModel(guid, "Title", "Label", new List<MediaPhotoItem> { }, new List<MediaVideoItem> { }, new List<MediaAudioItem> { }, new List<NewsItemModel> { }, DateTime.UtcNow, new List<string> { });
+        var archiveGrain = this._cluster.GrainFactory.GetGrain<IArchiveGrain>(guid);
+        await archiveGrain.CreateArchive(toSaveArchive);
 
+        var archiveInGrain = await this._cluster.GrainFactory.GetGrain<IArchiveGrain>(guid).Get();
 
+        Assert.NotNull(archiveInGrain);
     }
 
+    [Fact]
+    public async Task UpdateArchiveCorrectly()
+    {
+        var guid = Guid.NewGuid();
+
+        var toSaveArchive = new ArchiveModel(guid, "Title", "Label", new List<MediaPhotoItem> { }, new List<MediaVideoItem> { }, new List<MediaAudioItem> { }, new List<NewsItemModel> { }, DateTime.UtcNow, new List<string> { });
+        var archiveGrain = this._cluster.GrainFactory.GetGrain<IArchiveGrain>(guid);
+        await archiveGrain.CreateArchive(toSaveArchive);
+
+        var newerArchive = new ArchiveModel(guid, "Newer Title", "Newer Label", new List<MediaPhotoItem> { }, new List<MediaVideoItem> { }, new List<MediaAudioItem> { }, new List<NewsItemModel> { }, DateTime.UtcNow, new List<string> { });
+
+        await archiveGrain.Update(newerArchive);
+
+
+        var archiveInGrain = await this._cluster.GrainFactory.GetGrain<IArchiveGrain>(guid).Get();
+        Assert.NotEqual(toSaveArchive.Title, archiveInGrain.Title);
+        Assert.NotEqual(toSaveArchive.Label, archiveInGrain.Label);
+
+    }
+
+    [Fact]
+    public async Task CanDeleteArchiveCorrectly()
+    {
+        var guid = Guid.NewGuid();
+
+        var toSaveArchive = new ArchiveModel(guid, "Title", "Label", new List<MediaPhotoItem> { }, new List<MediaVideoItem> { }, new List<MediaAudioItem> { }, new List<NewsItemModel> { }, DateTime.UtcNow, new List<string> { });
+        var archiveGrain = this._cluster.GrainFactory.GetGrain<IArchiveGrain>(guid);
+        await archiveGrain.CreateArchive(toSaveArchive);
+
+        await this._cluster.GrainFactory.GetGrain<IArchiveGrain>(guid).Delete();
+        var archiveInGrain = await this._cluster.GrainFactory.GetGrain<IArchiveGrain>(guid).Get();
+
+        Assert.Equal("00000000-0000-0000-0000-000000000000", Convert.ToString(archiveInGrain.Id));
```

Comment:
Is het de bedoeling dat de ID empty is? lijkt me niet de bedoeling

Reply:
Ja dit hoort omdat als een item wordt verwijderd wordt de ID naar 0 gezet.